### ISSUE TITLE
Light vs. Dark mode map tile rendering issue

### ISSUE THREAD

**Issue by lasyakoechlin** [CartoDB/mobile-sdk#406]
I am having an issue with map tile rendering that seems to only occur with dark theme.

I have attached a gif file for download here, so that you can see 
https://drive.google.com/file/d/1rlkoHLloCd3PQNTqZg1eEuJkGpiWW4O7/view?usp=sharing

The above is on Apple and is using `Carto SDK 4.3.3`. I was also able to recreate the exact same behavior using Android with `Carto SDK 4.3.1`.

For some reason, in dark mode, those awkward polygons seem to show up and are visible whereas they do not appear in light mode.

The data is plotted using a `GeoJSONVectorTileDataSource` with `MBVectorTileDecoder` and `VectorTileLayer`. I have been able to confirm that those polygons do not exist in the source data whatsoever.

Thanks for your help!

**Comment by lasyakoechlin** [comment]
![image](https://user-images.githubusercontent.com/60749601/102823726-76681a00-43a9-11eb-9e67-c161ac3434a9.png)
![image](https://user-images.githubusercontent.com/60749601/102823749-7ff18200-43a9-11eb-9b2f-cda45bc0d5e8.png)

For easier visualization, here's a snapshot of the light vs dark.

Regardless of zoom level, the light never shows those weird discolored polygons. For dark mode, the polygons tend to change as the zoom changes. I suspect it has something to do with rendering of the map tiles?

**Comment by mtehver** [comment]
Thanks for reporting this. Could you make a small test with actual style and data demonstrating this? That would be help a lot.

**Comment by lasyakoechlin** [comment]
So, I've tried creating a simple sample app to demonstrate the problem but couldn't recreate it.

We were able to resolve the issue when we realized that two polygons (the building boundary and the hallway) were obviously overlapping but had different colors.

Changing the color codes in the CartoCSS to be the same resolved the issue.

So, my guess is that what was happening was that on a per-map-tile basis, depending on the user interactions, etc, perhaps one polygon was being rendered first/second and hence leading to a difference in which was visible.

In my carto css .json file, the layers were listed as:

```
"layers": [
    "indoor_text_labels",
    "indoor_above_main",
    "indoor_main",
    "indoor_pathway",
    "indoor_boundary"
  ],
```

where the  boundary polygon was assigned to the `indoor_boundary` layer and the hallway polygon was assigned to the `indoor_pathway` layer. So, I expected the behavior to be that the pathway/hallway would always be on top of the boundary. But, it's possible that sometimes that wasn't the case...?
